### PR TITLE
Tests cases for mutatingwebhookconfiguration update and create

### PR DESCRIPTION
**Pull request by fntlnz**
Signed-off-by: Lorenzo Fontana <lo@linux.com>

**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**: The `mutatingwebhookconfiguration` was completely uncovered, I'm adding a few simple cases I identified.

```release-note
NONE
```

### CODE REVIEW

File: pkg/registry/admissionregistration/mutatingwebhookconfiguration/strategy_test.go [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			m := mutatingWebhookConfigurationStrategy{
+				ObjectTyper:   nil,
+				NameGenerator: nil,
+			}
+			m.PrepareForCreate(tt.args.ctx, tt.args.obj)
+			v, ok := tt.args.obj.(*admissionregistration.MutatingWebhookConfiguration)
+			if !ok {
+				t.Errorf("PrepareForCreate: error casting runtime.Object to (*admissionregistration.MutatingWebhookConfiguration)")
```
Comment (neolit123) [inline pkg/registry/admissionregistration/mutatingwebhookconfiguration/strategy_test.go]: minor nit: `PrepareForCreate: ` should not be needed as go shows which test fails.
[1]


also, should be t.Fatalf() ?

File: pkg/registry/admissionregistration/mutatingwebhookconfiguration/strategy_test.go [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+			}
+			if v.Generation != tt.args.expectedGeneration {
+				t.Errorf("PrepareForCreate: generation number not matching, got: %d, want: %d", v.Generation, tt.args.expectedGeneration)
```
Comment (neolit123) [inline pkg/registry/admissionregistration/mutatingwebhookconfiguration/strategy_test.go]: [1]

File: pkg/registry/admissionregistration/mutatingwebhookconfiguration/strategy_test.go [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			m := mutatingWebhookConfigurationStrategy{
+				ObjectTyper:   nil,
+				NameGenerator: nil,
+			}
+			m.PrepareForUpdate(tt.args.ctx, tt.args.obj, tt.args.old)
+			v, ok := tt.args.obj.(*admissionregistration.MutatingWebhookConfiguration)
+			if !ok {
+				t.Errorf("PrepareForUpdate: error casting runtime.Object to (*admissionregistration.MutatingWebhookConfiguration)")
```
Comment (neolit123) [inline pkg/registry/admissionregistration/mutatingwebhookconfiguration/strategy_test.go]: [1]
also, this be Fatalf() ?

File: pkg/registry/admissionregistration/mutatingwebhookconfiguration/strategy_test.go [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+			}
+			if v.Generation != tt.args.expectedGeneration {
+				t.Errorf("PrepareForUpdate: generation number not matching, got: %d, want: %d", v.Generation, tt.args.expectedGeneration)
```
Comment (neolit123) [inline pkg/registry/admissionregistration/mutatingwebhookconfiguration/strategy_test.go]: [1]

**Review by neolit123 — COMMENTED** [review]
@fntlnz thanks for the PR.
added a couple of minor comments.

**Comment by neolit123** [comment]
@fntlnz 
> kube-apiserver: install webhook, after all equipement steps

the PR seems to be adding a unit test only, so possibly change the title and commit message?

/priority backlog

**Comment by fntlnz** [comment]
Thanks for your review @neolit123 - did the requested changes

**Comment by roycaihw** [comment]
cc @cheftako @jennybuckley

**Comment by fntlnz** [comment]
ok i'm giving up on this one.

**Pull request closed without merging**